### PR TITLE
Added __call() on QueryExpression to allow for and() and or() to be called

### PR DESCRIPTION
**Pull request by tigrang**
Copy/pasted straight from the referenced PR in #6477

### CODE REVIEW

File: src/Database/Expression/QueryExpression.php [hunk 1]
```diff
@@ -469,6 +469,22 @@ public function iterateParts(callable $callable)
...
+     * @param array $args The argumemts to pass to the method.
+     * @return \Cake\Database\Expression\QueryExpression
+     * @throws \BadMethodCallException
+     */
+    public function __call($method, $args)
+    {
+        if (in_array($method, ['and', 'or'])) {
+            return call_user_func_array([$this, $method . '_'], $args);
+        }
+        throw new \BadMethodCallException;
```
Comment (ADmad) [inline src/Database/Expression/QueryExpression.php]: Including an exception message stating the invalid method name would be nice. Makes debugging easier.

**Comment by markstory** [comment]
Looks good to me. :+1:

**Comment by lorenzo** [comment]
Can you target this to master instead?

**Comment by tigrang** [comment]
See #6624

**Pull request closed without merging**